# Flight Price Prediction - EDA and Feature Engineering

## Imports

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

from sklearn.preprocessing import LabelEncoder

## Reading the Dataset

In [18]:
%pwd

'd:\\Projects\\Exploratory Data Analysis\\Flight Price Predcition\\notebook'

In [19]:
df_train = pd.read_excel('../data/Data_Train.xlsx')
df_test = pd.read_excel('../data/Test_set.xlsx')

In [20]:
df_train.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

In [21]:
df_test.head()

Airline Date_of_Journey    Source Destination            Route  \
0        Jet Airways       6/06/2019     Delhi      Cochin  DEL → BOM → COK   
1             IndiGo      12/05/2019   Kolkata    Banglore  CCU → MAA → BLR   
2        Jet Airways      21/05/2019     Delhi      Cochin  DEL → BOM → COK   
3  Multiple carriers      21/05/2019     Delhi      Cochin  DEL → BOM → COK   
4           Air Asia      24/06/2019  Banglore       Delhi        BLR → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops              Additional_Info  
0    17:30  04:25 07 Jun  10h 55m      1 stop                      No info  
1    06:20         10:20       4h      1 stop                      No info  
2    19:15  19:00 22 May  23h 45m      1 stop  In-flight meal not included  
3    08:00         21:00      13h      1 stop                      No info  
4    23:55  02:45 25 Jun   2h 50m    non-stop                      No info

## Exploring Dataset

In [22]:
df_train.columns

Index(['Airline', 'Date_of_Journey', 'Source', 'Destination', 'Route',
       'Dep_Time', 'Arrival_Time', 'Duration', 'Total_Stops',
       'Additional_Info', 'Price'],
      dtype='object')

In [23]:
df_test.columns

Index(['Airline', 'Date_of_Journey', 'Source', 'Destination', 'Route',
       'Dep_Time', 'Arrival_Time', 'Duration', 'Total_Stops',
       'Additional_Info'],
      dtype='object')

In [24]:
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10683 entries, 0 to 10682
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10683 non-null  object
 1   Date_of_Journey  10683 non-null  object
 2   Source           10683 non-null  object
 3   Destination      10683 non-null  object
 4   Route            10682 non-null  object
 5   Dep_Time         10683 non-null  object
 6   Arrival_Time     10683 non-null  object
 7   Duration         10683 non-null  object
 8   Total_Stops      10682 non-null  object
 9   Additional_Info  10683 non-null  object
 10  Price            10683 non-null  int64 
dtypes: int64(1), object(10)
memory usage: 918.2+ KB


In [25]:
df_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2671 entries, 0 to 2670
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          2671 non-null   object
 1   Date_of_Journey  2671 non-null   object
 2   Source           2671 non-null   object
 3   Destination      2671 non-null   object
 4   Route            2671 non-null   object
 5   Dep_Time         2671 non-null   object
 6   Arrival_Time     2671 non-null   object
 7   Duration         2671 non-null   object
 8   Total_Stops      2671 non-null   object
 9   Additional_Info  2671 non-null   object
dtypes: object(10)
memory usage: 208.8+ KB


In [26]:
df_train.describe()

Price
count  10683.000000
mean    9087.064121
std     4611.359167
min     1759.000000
25%     5277.000000
50%     8372.000000
75%    12373.000000
max    79512.000000

In [27]:
df_test.describe()

Airline Date_of_Journey Source Destination            Route  \
count          2671            2671   2671        2671             2671   
unique           11              44      5           6              100   
top     Jet Airways       9/05/2019  Delhi      Cochin  DEL → BOM → COK   
freq            897             144   1145        1145              624   

       Dep_Time Arrival_Time Duration Total_Stops Additional_Info  
count      2671         2671     2671        2671            2671  
unique      199          704      320           5               6  
top       10:00        19:00   2h 50m      1 stop         No info  
freq         62          113      122        1431            2148

In [28]:
df_train.shape

(10683, 11)

In [29]:
df_test.shape

(2671, 10)

In [30]:
## Merge Train and Test DataFrames
df = pd.concat([df_train, df_test], ignore_index=True)
df.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info    Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897.0  
1    05:50         13:15   7h 25m     2 stops         No info   7662.0  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882.0  
3    18:05         23:30   5h 25m      1 stop         No info   6218.0  
4    16:50         21:35   4h 45m      1 stop         No info  13302.0

In [31]:
df.tail()

Airline Date_of_Journey   Source Destination  \
13349          Air India       6/06/2019  Kolkata    Banglore   
13350             IndiGo      27/03/2019  Kolkata    Banglore   
13351        Jet Airways       6/03/2019    Delhi      Cochin   
13352          Air India       6/03/2019    Delhi      Cochin   
13353  Multiple carriers      15/06/2019    Delhi      Cochin   

                 Route Dep_Time  Arrival_Time Duration Total_Stops  \
13349  CCU → DEL → BLR    20:30  20:25 07 Jun  23h 55m      1 stop   
13350        CCU → BLR    14:20         16:55   2h 35m    non-stop   
13351  DEL → BOM → COK    21:50  04:25 07 Mar   6h 35m      1 stop   
13352  DEL → BOM → COK    04:00         19:15  15h 15m      1 stop   
13353  DEL → BOM → COK    04:55         19:15  14h 20m      1 stop   

      Additional_Info  Price  
13349         No info    NaN  
13350         No info    NaN  
13351         No info    NaN  
13352         No info    NaN  
13353         No info    NaN

### Checking values in each attribute and converting them to proper datatype

#### Airline

In [32]:
df['Airline'].value_counts()

Airline
Jet Airways                          4746
IndiGo                               2564
Air India                            2192
Multiple carriers                    1543
SpiceJet                             1026
Vistara                               608
Air Asia                              405
GoAir                                 240
Multiple carriers Premium economy      16
Jet Airways Business                    8
Vistara Premium economy                 5
Trujet                                  1
Name: count, dtype: int64

In [33]:
df['Airline'] = df['Airline'].astype('str')

#### Date of Journey

In [34]:
df['Date_of_Journey'].value_counts()

Date_of_Journey
18/05/2019    633
6/06/2019     630
9/05/2019     628
12/06/2019    628
21/05/2019    615
9/06/2019     614
21/03/2019    516
15/05/2019    511
27/05/2019    447
24/06/2019    436
15/06/2019    433
1/06/2019     430
27/06/2019    430
3/06/2019     425
6/03/2019     405
24/03/2019    401
27/03/2019    364
24/05/2019    357
6/05/2019     355
1/05/2019     339
1/04/2019     335
12/05/2019    327
3/03/2019     280
9/03/2019     255
18/03/2019    197
15/03/2019    195
01/03/2019    186
12/03/2019    185
9/04/2019     149
3/04/2019     138
21/06/2019    133
18/06/2019    127
09/03/2019    123
03/03/2019    123
06/03/2019    123
3/05/2019     117
6/04/2019     114
24/04/2019    113
15/04/2019    112
27/04/2019    109
21/04/2019    104
18/04/2019     79
12/04/2019     74
1/03/2019      59
Name: count, dtype: int64

In [35]:
df['Journey_Date'] = df["Date_of_Journey"].apply(lambda x: x.split('/')[0])
df['Journey_Month'] = df["Date_of_Journey"].apply(lambda x: x.split('/')[1])
df['Journey_Year'] = df["Date_of_Journey"].apply(lambda x: x.split('/')[2])

In [36]:
df['Journey_Date'] = df['Journey_Date'].astype('int')
df['Journey_Month'] = df['Journey_Month'].astype('int')
df['Journey_Year'] = df['Journey_Year'].astype('int')

In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13354 entries, 0 to 13353
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Airline          13354 non-null  object 
 1   Date_of_Journey  13354 non-null  object 
 2   Source           13354 non-null  object 
 3   Destination      13354 non-null  object 
 4   Route            13353 non-null  object 
 5   Dep_Time         13354 non-null  object 
 6   Arrival_Time     13354 non-null  object 
 7   Duration         13354 non-null  object 
 8   Total_Stops      13353 non-null  object 
 9   Additional_Info  13354 non-null  object 
 10  Price            10683 non-null  float64
 11  Journey_Date     13354 non-null  int64  
 12  Journey_Month    13354 non-null  int64  
 13  Journey_Year     13354 non-null  int64  
dtypes: float64(1), int64(3), object(10)
memory usage: 1.4+ MB


In [38]:
df.drop('Date_of_Journey',axis=1,inplace=True)

#### Source and Destination

In [39]:
df['Source'].value_counts()

Source
Delhi       5682
Kolkata     3581
Banglore    2752
Mumbai       883
Chennai      456
Name: count, dtype: int64

In [40]:
df['Destination'].value_counts()

Destination
Cochin       5682
Banglore     3581
Delhi        1582
New Delhi    1170
Hyderabad     883
Kolkata       456
Name: count, dtype: int64

In [41]:
df['Source'] = df['Source'].astype('str')
df['Destination'] = df['Destination'].astype('str')

In [42]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13354 entries, 0 to 13353
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Airline          13354 non-null  object 
 1   Source           13354 non-null  object 
 2   Destination      13354 non-null  object 
 3   Route            13353 non-null  object 
 4   Dep_Time         13354 non-null  object 
 5   Arrival_Time     13354 non-null  object 
 6   Duration         13354 non-null  object 
 7   Total_Stops      13353 non-null  object 
 8   Additional_Info  13354 non-null  object 
 9   Price            10683 non-null  float64
 10  Journey_Date     13354 non-null  int64  
 11  Journey_Month    13354 non-null  int64  
 12  Journey_Year     13354 non-null  int64  
dtypes: float64(1), int64(3), object(9)
memory usage: 1.3+ MB


In [43]:
df.drop('Route',axis=1,inplace=True)

#### Duration

In [44]:
df['Duration'].value_counts()

Duration
2h 50m     672
1h 30m     493
2h 45m     432
2h 55m     418
2h 35m     399
          ... 
3h 45m       1
31h 10m      1
40h 40m      1
22h 50m      1
32h 25m      1
Name: count, Length: 374, dtype: int64

In [45]:
df['Duration'].str.split(' ').str[0]

0         2h
1         7h
2        19h
3         5h
4         4h
        ... 
13349    23h
13350     2h
13351     6h
13352    15h
13353    14h
Name: Duration, Length: 13354, dtype: object

In [46]:
df['Duration_Hour'] = df['Duration'].str.split(' ').str[0].str.replace('h','')

In [47]:
df['Duration_Min'] = df['Duration'].str.split(' ').str[1].str.replace('m','')

In [48]:
df.drop('Duration',axis=1,inplace=True)

In [49]:
#df.loc[[df['Duration_Hour']=='5m'], 'Duration_Min'] = '5'
#df.loc[[df['Duration_Hour']=='5m'], 'Duration_Hour'] = '0'
mask = df['Duration_Hour'].eq('5m')
df.loc[mask, ['Duration_Min', 'Duration_Hour']] = ['5', '0']


In [50]:
df['Duration_Min'].fillna(0,inplace=True)

C:\Users\Akhil\AppData\Local\Temp\ipykernel_18868\367320920.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Duration_Min'].fillna(0,inplace=True)


In [51]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13354 entries, 0 to 13353
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Airline          13354 non-null  object 
 1   Source           13354 non-null  object 
 2   Destination      13354 non-null  object 
 3   Dep_Time         13354 non-null  object 
 4   Arrival_Time     13354 non-null  object 
 5   Total_Stops      13353 non-null  object 
 6   Additional_Info  13354 non-null  object 
 7   Price            10683 non-null  float64
 8   Journey_Date     13354 non-null  int64  
 9   Journey_Month    13354 non-null  int64  
 10  Journey_Year     13354 non-null  int64  
 11  Duration_Hour    13354 non-null  object 
 12  Duration_Min     13354 non-null  object 
dtypes: float64(1), int64(3), object(9)
memory usage: 1.3+ MB


In [52]:
df['Duration_Hour'] = df['Duration_Hour'].astype('int')
df['Duration_Min'] = df['Duration_Min'].astype('int')

In [53]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13354 entries, 0 to 13353
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Airline          13354 non-null  object 
 1   Source           13354 non-null  object 
 2   Destination      13354 non-null  object 
 3   Dep_Time         13354 non-null  object 
 4   Arrival_Time     13354 non-null  object 
 5   Total_Stops      13353 non-null  object 
 6   Additional_Info  13354 non-null  object 
 7   Price            10683 non-null  float64
 8   Journey_Date     13354 non-null  int64  
 9   Journey_Month    13354 non-null  int64  
 10  Journey_Year     13354 non-null  int64  
 11  Duration_Hour    13354 non-null  int64  
 12  Duration_Min     13354 non-null  int64  
dtypes: float64(1), int64(5), object(7)
memory usage: 1.3+ MB


#### Departure Time and Arrival Time to Hours and Minutes

In [54]:
df['Dep_Time'] = pd.to_datetime(df['Dep_Time'])
df['Arrival_Time'] = pd.to_datetime(df['Arrival_Time'])

C:\Users\Akhil\AppData\Local\Temp\ipykernel_18868\252221560.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['Dep_Time'] = pd.to_datetime(df['Dep_Time'])
C:\Users\Akhil\AppData\Local\Temp\ipykernel_18868\252221560.py:2: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['Arrival_Time'] = pd.to_datetime(df['Arrival_Time'])


In [55]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13354 entries, 0 to 13353
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   Airline          13354 non-null  object        
 1   Source           13354 non-null  object        
 2   Destination      13354 non-null  object        
 3   Dep_Time         13354 non-null  datetime64[ns]
 4   Arrival_Time     13354 non-null  datetime64[ns]
 5   Total_Stops      13353 non-null  object        
 6   Additional_Info  13354 non-null  object        
 7   Price            10683 non-null  float64       
 8   Journey_Date     13354 non-null  int64         
 9   Journey_Month    13354 non-null  int64         
 10  Journey_Year     13354 non-null  int64         
 11  Duration_Hour    13354 non-null  int64         
 12  Duration_Min     13354 non-null  int64         
dtypes: datetime64[ns](2), float64(1), int64(5), object(5)
memory usage: 1.3+ MB


In [56]:
df['Dep_Hour'] = df['Dep_Time'].dt.hour
df['Dep_Min'] = df['Dep_Time'].dt.minute
df['Arrival_Hour'] = df['Arrival_Time'].dt.hour
df['Arrival_Min'] = df['Arrival_Time'].dt.minute

In [57]:
df.drop(['Dep_Time','Arrival_Time'],axis=1,inplace=True)

In [58]:
#df[['Dep_Time', 'Arrival_Time']]

In [59]:
#df['Arrival_Time'] = df['Arrival_Time'].str.split(' ').str[0]

In [60]:
#df['Dep_Hour'] = df['Dep_Time'].str.split(':').str[0]
#df['Dep_Min'] = df['Dep_Time'].str.split(':').str[1]

In [61]:
#df['Arrival_Hour'] = df['Arrival_Time'].str.split(':').str[0]
#df['Arrival_Min'] = df['Arrival_Time'].str.split(':').str[1]

In [62]:
#df.drop(['Dep_Hour','Dep_Min','Arrival_Hour','Arrival_Min'],axis=1,inplace=True)

In [63]:
#df.info()

#### Total Stops

In [64]:
df.isnull().sum()

Airline               0
Source                0
Destination           0
Total_Stops           1
Additional_Info       0
Price              2671
Journey_Date          0
Journey_Month         0
Journey_Year          0
Duration_Hour         0
Duration_Min          0
Dep_Hour              0
Dep_Min               0
Arrival_Hour          0
Arrival_Min           0
dtype: int64

In [65]:
df[df['Total_Stops'].isnull()]

Airline Source Destination Total_Stops Additional_Info   Price  \
9039  Air India  Delhi      Cochin         NaN         No info  7480.0   

      Journey_Date  Journey_Month  Journey_Year  Duration_Hour  Duration_Min  \
9039             6              5          2019             23            40   

      Dep_Hour  Dep_Min  Arrival_Hour  Arrival_Min  
9039         9       45             9           25

In [66]:
df['Total_Stops'].fillna('non-stop',inplace=True)

C:\Users\Akhil\AppData\Local\Temp\ipykernel_18868\2176209460.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Total_Stops'].fillna('non-stop',inplace=True)


In [67]:
df['Total_Stops'].value_counts()

Total_Stops
1 stop      7056
non-stop    4341
2 stops     1899
3 stops       56
4 stops        2
Name: count, dtype: int64

In [68]:
df['Total_Stops'] = df['Total_Stops'].map({'non-stop':0, '1 stop':1, '2 stops':2, '3 stops':3, '4 stops':4})

In [69]:
df['Total_Stops'].astype('int')

0        0
1        2
2        2
3        1
4        1
        ..
13349    1
13350    0
13351    1
13352    1
13353    1
Name: Total_Stops, Length: 13354, dtype: int64

In [70]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13354 entries, 0 to 13353
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Airline          13354 non-null  object 
 1   Source           13354 non-null  object 
 2   Destination      13354 non-null  object 
 3   Total_Stops      13354 non-null  int64  
 4   Additional_Info  13354 non-null  object 
 5   Price            10683 non-null  float64
 6   Journey_Date     13354 non-null  int64  
 7   Journey_Month    13354 non-null  int64  
 8   Journey_Year     13354 non-null  int64  
 9   Duration_Hour    13354 non-null  int64  
 10  Duration_Min     13354 non-null  int64  
 11  Dep_Hour         13354 non-null  int32  
 12  Dep_Min          13354 non-null  int32  
 13  Arrival_Hour     13354 non-null  int32  
 14  Arrival_Min      13354 non-null  int32  
dtypes: float64(1), int32(4), int64(6), object(4)
memory usage: 1.3+ MB


#### Final Sanity Check

In [71]:
df.isnull().sum()

Airline               0
Source                0
Destination           0
Total_Stops           0
Additional_Info       0
Price              2671
Journey_Date          0
Journey_Month         0
Journey_Year          0
Duration_Hour         0
Duration_Min          0
Dep_Hour              0
Dep_Min               0
Arrival_Hour          0
Arrival_Min           0
dtype: int64

In [72]:
labelencoder=LabelEncoder()

df['Airline']=labelencoder.fit_transform(df['Airline'])
#df['Source']=labelencoder.fit_transform(df['Source'])
#df['Destination']=labelencoder.fit_transform(df['Destination'])
df['Additional_Info']=labelencoder.fit_transform(df['Additional_Info'])

In [73]:
df.head()

Airline    Source Destination  Total_Stops  Additional_Info    Price  \
0        3  Banglore   New Delhi            0                8   3897.0   
1        1   Kolkata    Banglore            2                8   7662.0   
2        4     Delhi      Cochin            2                8  13882.0   
3        3   Kolkata    Banglore            1                8   6218.0   
4        3  Banglore   New Delhi            1                8  13302.0   

   Journey_Date  Journey_Month  Journey_Year  Duration_Hour  Duration_Min  \
0            24              3          2019              2            50   
1             1              5          2019              7            25   
2             9              6          2019             19             0   
3            12              5          2019              5            25   
4             1              3          2019              4            45   

   Dep_Hour  Dep_Min  Arrival_Hour  Arrival_Min  
0        22       20             1           10  
1         5       50            13           15  
2         9       25             4           25  
3        18        5            23           30  
4        16       50            21           35

Source
*Delhi       5682
*Kolkata     3581
*Banglore    2752
*Mumbai       883
*Chennai      456

Destination
*Cochin       5682
*Banglore     3581
*Delhi        1582
*New Delhi    1170
*Hyderabad     883
*Kolkata       456

Banglore 1
Chennai 2
Cochin 3
Delhi, New Delhi 4
Hyderabad 5
Kolkata 6 
Mumbai 7


In [74]:
df['Source'] = df['Source'].map({'Banglore':1, 'Chennai':2, 'Cochin':3, 'Delhi':4, 'New Delhi':4, 'Hyderabad':5, 'Kolkata':6, 'Mumbai':7})
df['Destination'] = df['Destination'].map({'Banglore':1, 'Chennai':2, 'Cochin':3, 'Delhi':4, 'New Delhi':4, 'Hyderabad':5, 'Kolkata':6, 'Mumbai':7})

In [75]:
df.head()

Airline  Source  Destination  Total_Stops  Additional_Info    Price  \
0        3       1            4            0                8   3897.0   
1        1       6            1            2                8   7662.0   
2        4       4            3            2                8  13882.0   
3        3       6            1            1                8   6218.0   
4        3       1            4            1                8  13302.0   

   Journey_Date  Journey_Month  Journey_Year  Duration_Hour  Duration_Min  \
0            24              3          2019              2            50   
1             1              5          2019              7            25   
2             9              6          2019             19             0   
3            12              5          2019              5            25   
4             1              3          2019              4            45   

   Dep_Hour  Dep_Min  Arrival_Hour  Arrival_Min  
0        22       20             1           10  
1         5       50            13           15  
2         9       25             4           25  
3        18        5            23           30  
4        16       50            21           35

In [76]:
df['Additional_Info'].value_counts()

Additional_Info
8    10493
5     2426
7      396
0       20
4        8
3        5
6        3
1        1
9        1
2        1
Name: count, dtype: int64